lib/Github/HttpClient/CachedHttpClient.php

```diff
@@ -27,6 +27,13 @@ class CachedHttpClient extends HttpClient
     private $lastCachedResponse;
 
     /**
+     * $path + query parameter(s) if they exist.
+     *
+     * @var string
+     */
+    private $path;
```

Comment:
please don't store this in a property. The more state is added in the class, the more complex reasonning about it is (I already think that having ``getLastResponse`` in the ``HttpClientInterface`` is a big mistake).